DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h

```diff
@@ -110,7 +109,13 @@ class SiPixelRecHitQuality {
     //------------------------------------------------------
     //
     inline void setProbabilityXY(float prob, QualWordType& qualWord) const {
-      if (prob < 0 || prob > 1) {
+      if (prob < 0.0f) {
+        warningOutOfBoundProb("XY", prob, qualWord);
+        prob = 0;
+      } else if (prob > 1.0f && prob < 1.0f + 32 * std::numeric_limits<float>::epsilon()) {
```

Comment:
is this really justifiable to go up to 32 ? The 32 seems to be ad-hoc (why not 42?)
The only reason I brought it up was in the connection of the printout.

@OzAmram @pmaksim1 please clarify the domain of the functions calling these `setProbability*` methods updated in this PR.

Reply:
The probx and proby are bounded between probmin (which is 1e-16) [(1)](https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc#L172) and 1 [(2)](https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco.cc#L1154).
Then the ProbXY gets computed as probx * proby * (1 - log(probx * proby) [(3)](https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelRecHits/src/PixelCPEBase.cc#L448-L449) which should also be bounded in the same range but can introduce some rounding errors